proposals/custom-descriptors/Overview.md

```diff
@@ -562,164 +559,254 @@ Furthermore, the design goals are to be:
+constructorconfig ::= constructorname:name
+                      vec(methodconfig)
 
+parentidx ::= u32               
 ```
 
-The descriptors custom section starts with a version number,
-which is intended to help tools and engines manage
-backward-incompatible changes to this custom section format
-during proposal development.
-It will be removed from the final version of the custom section.
+If any of the arguments are null,
+of if the data does not conform to this grammar,
+or if the total number of `protoconfig`s does not match the length of the prototypes array,
+or if the total number of `methodconfig`s and `constructorconfig`s does not match the length of the functions array,
+then `configureAll` throws an error.
 
-Next comes `modulename`, which serves two purposes.
-It is the module name used to look up user-provided prototypes on the import object
-for use in configuring `DescriptorOptions` objects,
-and it is also the module name where the configured `DescriptorOptions`
-are written back to the import object to be looked up by core instantiation.
-A module may import configured `DescriptorOptions` values
-from multiple different module names
-by including multiple descriptors sections.
+> TODO: Detail the thrown error.
 
-Following the `modulename` is a sequence of `descriptorentry`,
-each of which describes a declared `DescriptorOptions` value or an imported prototype
-for use in the prototype chain of subsequent declared `DescriptorOptions`.
-A declaration can optionally specify the index of a previous entry
-to provide the parent in the configured prototype chain.
-If the previous entry is an `importentry`, the `modulename` and `importname`
-are used to look up the prototype in the imports object.
-Otherwise, if the previous entry is a `declentry`,
-the prototype is looked up from the declared `DescriptorOptions`.
-If there is no configured parent,
-the parent of the new prototype is `Object.prototype`.
+After parsing the configuration data,
+`configureAll` iterates through the `protoconfigs`.
+Each `protoconfig` corresponds to a prototype in the prototypes list.
+For each `methodconfig` in the `protoconfig`,
+the next function is taken from the functions array,
+wrapped to take its receiver as its first parameter,
+and installed as a method, getter, or setter with the given name on the current prototype.
 
-Each declared descriptor has a vector of export names.
-After the descriptor is created,
-it is written into the import object under `modulename` with each of these export names.
-The core Wasm module will then import the descriptors using these names.
-Typically there will be no more than one name in the vector.
-The vector may be empty if the descriptor only exists
-to declaratively populate links in the prototype chain used by other descriptors.
+If the `protoconfig` has a `constructorconfig`,
+the next function from the functions array is wrapped to be callable with `new`,
```

Comment:
It's not clear whether "callable with `new`" means that it *has to* be called with `new` (like a JS `class` constructor), or that `new` is allowed (like a JS `function`).

More generally, does it behave like a `class` constructor or like a `function`? (regular Wasm exported functions behave like `=>` functions)

Reply:
Yes, I think that makes sense. I updated the text to clarify that the wrapper can _optionally_ be called with `new`.